packages/multichain/src/middlewares/MultichainSubscriptionManager.ts

```diff
@@ -108,10 +112,17 @@ export class MultichainSubscriptionManager extends SafeEventEmitter {
       },
     );
 
-    this.#subscriptions.push({
+    const newSubscriptionManagerEntry = {
       ...subscriptionKey,
       subscriptionManager,
-    });
+    };
+    subscriptionManager.destroy = subscriptionManager.middleware.destroy;
+    subscriptionManager.middleware.destroy = this.#unsubscribe.bind(
+      this,
+      newSubscriptionManagerEntry,
+    );
```

Comment:
🤔 maybe I'm missing it but any reason we could do this instead?
```suggestion
    subscriptionManager.destroy = this.#unsubscribe.bind(
      this,
      newSubscriptionManagerEntry,
    );
```

Reply:
https://github.com/MetaMask/eth-json-rpc-filters/blob/main/subscriptionManager.js#L24

There's two issues: 

* `SubscriptionManager.middleware.destroy` is set to the `SubscriptionManager.destroy` function (as opposed to being a function that calls `SubscriptionManager.destroy`)
* `SubscriptionManager.destroy` is not actually exported/exposed and only accessible through `SubscriptionManager.middleware.destroy`

Reply:
oh wow I totally misread that